pull_request_template.md

```diff
@@ -0,0 +1,28 @@
+## Description
+{ This PR adds ... }
+
+{ This PR fixes ... }
+
+{ This PR removes ... }
+
+## Due Diligence Checklist
+- [ ] Tests
+- [ ] Documentation
+- [ ] Consolidated commits
```

Comment:
Good question -- this could mean different levels of "consolidated" and I'm curious what folks prefer.

Question 1: Would you be cool with an item along the lines of "are your commit messages well written / are the commits well formed" or would you rather keep commit structure out of the checklist entirely.

Question 2: As far as squashing -- right, this template came from a projects where PR feedback was expected to be consolidated into the original commits (I believe the benefit was that the final code maintains the context of the commit / commit message, rather than having some of it appear in a "pr feedback" commit).

Functionally this means `git rebase -i` and marking fixups, and then force pushing to the feature branch.

The checklist item as written is definitely referencing that second approach.

Does that clarify at all?  What do you think about those two types after hearing the intent?

Reply:
Commit messages and structure are a key piece of technical communication, and I strongly believe we should be practicing that as developers and looking for that in code review. Branches should be rewritten to address concerns brought up in PRs; +1 to what @slifty is saying about avoiding "pr feedback" commits.

I don't want us to squash meaningfully distinct commits into one big jumble, and I especially don't want to use GitHub's squash-and-merge button feature.

If we have a checklist like this, I think this entry belongs on it. I'd be open to dropping the checklist as a whole, but I think each item on it is a practice I'd like for us to commit to and to look for during code reviews.

Reply:
I took this and the rest of the checklist out; we can always add it in later if we start getting public contributions (it strikes me that these kinds of checklists are probably more useful for community contributors and less useful for core contributors) glad it was there to prompt this discussion though.